MiddleClick/Controller.m

```diff
@@ -263,19 +263,25 @@ int touchCallback(int device, Finger* data, int nFingers, double timestamp,
     }
     
     if (nFingers == fingersQua) {
-      Finger* f1 = &data[0];
-      Finger* f2 = &data[1];
-      Finger* f3 = &data[2];
-      
       if (maybeMiddleClick == YES) {
-        middleclickX = (f1->normalized.pos.x + f2->normalized.pos.x + f3->normalized.pos.x);
-        middleclickY = (f1->normalized.pos.y + f2->normalized.pos.y + f3->normalized.pos.y);
+        for (int i = 0; i < fingersQua; i++)
+        {
+          mtPoint pos = ((Finger *)&data[i])->normalized.pos;
+          middleclickX += pos.x;
```

Comment:
No need to set middleclickX and middleclickY to 0 explicitly since they are set to 0 at l.249~250.